### DISCUSSION TITLE
Add an unlock feature for Zappi

### DISCUSSION THREAD

**Discussion by JosyBan** [CJNE/ha-myenergi#493] [Ideas]
There is an unlock call available in the pymyenergi project for the zappi.
https://github.com/CJNE/pymyenergi/blob/c4b899eaffcf0594b6ed66ea0a68ba6aa8547764/pymyenergi/zappi.py#L363

This feature could be exposed in ha-myenergi as it would be very useful

**Comment by CosAdrPala** [comment]
Any news on the unlock function?

**Reply by JosyBan** [reply]
This was done last March for the Zappi. https://github.com/CJNE/ha-myenergi/pull/494

**Reply by CosAdrPala** [reply]
Thank you for your answer, but in HA I can't seem to find the unlock service or entity. If you integrated this, can you please share your example?

**Reply by LeiChat** [reply]
@CosAdrPala It's an action on myenergi that can target a zappi device.

```
action: myenergi.myenergi_unlock
metadata: {}
data: {}
target:
  device_id: YOUR-ZAPPI-DEVICE-ID
```
![image](https://github.com/user-attachments/assets/194904fb-da45-4aff-885a-c6a7cd84a718)

![image](https://github.com/user-attachments/assets/d3e7e2e3-f145-4d67-b563-b488c25647f6)

**Comment by JosyBan** [comment]
This is completed.  https://github.com/CJNE/ha-myenergi/pull/494

**Comment by G6EJD** [comment]
And the sponsor / creator was so interested they never knew! Can close the issue now...

**Comment by JosyBan** [comment]
@G6EJD I did the work, your welcome!!
